nuxeo-java-client/src/main/java/org/nuxeo/client/api/objects/Document.java

```diff
@@ -258,6 +267,9 @@ public String getName() {
     }
 
     public void set(String key, Object value) {
+        if (value instanceof Calendar || value instanceof Date) {
+            throw new IllegalArgumentException(MSG_DATE_UNSUPPORTED);
+        }
```

Comment:
For the duplication, I suggest to `@Deprecate` one of the getter / setter pairs:

- `getPropertyValue(String)` / `setPropertyValue(String, Object)`
- `get(String)` / `set(String, Object)`

The chosen method will be deprecated for the next release R, then will be removed for the release R+1. Therefore, users will have time to migrate their code. Here's the ticket I've just created: [JAVACLIENT-121](https://jira.nuxeo.com/browse/JAVACLIENT-121).